python-lib/dku_utils/node_pool.py

```diff
@@ -25,7 +27,84 @@ def get_node_pool_args(node_pool):
         args = args + ['--managed', '--spot']
 
     if len(node_pool.get('publicKeyName', '')) > 0:
-        args = args + ["--ssh-access"]
+        args = args + ['--ssh-access']
         args = args + ['--ssh-public-key', node_pool.get('publicKeyName', '')]
-        
-    return args
\ No newline at end of file
+
+    node_pool['labels'] = node_pool.get('labels', {})
+    if bool(node_pool['labels']):
+        labels = []
+        for label_key, label_value in node_pool['labels'].items():
+            if not label_key:
+                raise Exception('At least one node pool label key is not valid, please ensure label keys are not empty. Observed labels: %s' % node_pool['labels'])
+            if not label_value:
+                label_value = ''
+            labels.append('%s=%s' % (label_key, label_value))
+        args += ['--node-labels', ','.join(labels)]
+
+    return args
+
+def get_node_pool_yaml(node_pool, networking_settings):
+    yaml = {
+        'iam': {
+            'withAddonPolicies': {
+                'imageBuilder': True # Adding full ECR access to the node group
+            }
+        }
+    }
+    if 'machineType' in node_pool:
+        yaml['instanceType'] = node_pool['machineType']
+    yaml['volumeType'] = node_pool.get('diskType', 'gp2')
+    if 'diskSizeGb' in node_pool and node_pool['diskSizeGb'] > 0:
+        yaml['volumeSize'] = node_pool['diskSizeGb']
+    else:
+        yaml['volumeSize'] = 200 # also defined as default value in parameter-sets/node-pool-request/parameter-set.json
+
+    yaml['desiredCapacity'] = node_pool.get('numNodes', 3)
+    if node_pool.get('numNodesAutoscaling', False):
+        yaml['iam']['withAddonPolicies']['autoScaler'] = True
+        yaml['minSize'] = node_pool.get('minNumNodes', 2)
+        yaml['maxSize'] = node_pool.get('maxNumNodes', 5)
+
+    yaml['tags'] = node_pool.get('tags', {})
+    yaml['taints'] = build_node_pool_taints_yaml(node_pool)
+    node_pool['labels'] = node_pool.get('labels', {})
+    if any(_is_none_or_blank(label_key) for label_key in node_pool['labels'].keys()):
+        raise Exception('At least one node pool label key is not valid, please ensure label keys are not empty. Observed labels: %s' % node_pool['labels'])
+    yaml['labels'] = node_pool['labels']
+    yaml['spot'] = node_pool.get('useSpotInstances', False)
+
+    sshPublicKeyName = node_pool.get('publicKeyName', '')
+    if not _is_none_or_blank(sshPublicKeyName):
+        yaml['ssh'] = {
+            'allow': True,
+            'publicKeyName': sshPublicKeyName
+        }
+
+    if len(networking_settings.get('securityGroups', [])) > 0:
+        yaml['securityGroups'] = {
+            'attachIDs': networking_settings['securityGroups']
```

Comment:
I'll trim some other strings such as subnet ids because it seems like a weird pain point. The spaces are not shown in the UI, but the cluster start fails with weird errors.